### PR TITLE
updated autocomplete search behavior

### PR DESCRIPTION
**Pull request by ksantiagoBAH**
## Description
As a comparison tool user, I need the text search auto complete to just populate the text box so that I have a chance to review my inputs prior to moving on to the next page.## Testing done
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/12290

## Testing
local testing

## Acceptance criteria
- [x] After selecting a school in the "Enter a school, location, or employer name" field on the CT Landing Page, the field is populated with the selected school name.
a. The user remains on the Comparison Tool Landing Page
b. The user is not redirected to the Comparison Tool Search Results Page.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/gi/components/search/KeywordSearch.jsx [hunk 1]
```diff
@@ -54,7 +54,11 @@ export class KeywordSearch extends React.Component {
       event: 'gibct-autosuggest',
       'gibct-autosuggest-value': searchQuery,
     });
-    this.props.onFilterChange(searchQuery);
+    if (searchQuery) {
```
Comment (zurbergram) [inline src/applications/gi/components/search/KeywordSearch.jsx:57]: Why are checking on the value of `searchQuery`?

Why not have this just be behavior that is enabled by setting a property to true?

File: src/applications/gi/components/search/KeywordSearch.jsx [hunk 2]
```diff
@@ -54,7 +54,11 @@ export class KeywordSearch extends React.Component {
...
+      this.props.onUpdateAutocompleteSearchTerm(searchQuery);
+    } else {
+      this.props.onFilterChange(searchQuery);
```
Comment (zurbergram) [inline src/applications/gi/components/search/KeywordSearch.jsx:60]:
```suggestion
      this.props.onFilterChange('name', searchQuery);
```

File: src/applications/gi/components/search/KeywordSearch.jsx (hunk 1 above)
Comment (zurbergram) [inline src/applications/gi/components/search/KeywordSearch.jsx:57]: Will the Search Page continue to behave as expected with this logic?

Comment (ksantiagomorgan6) [reply src/applications/gi/components/search/KeywordSearch.jsx:57]: It passed local testing the way I had it with the expected behavior.

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: src/applications/gi/containers/LandingPage.jsx [hunk 3]
```diff
@@ -177,6 +178,7 @@ export class LandingPage extends React.Component {
               {!isVetTecSelected(this.props.filters) && (
                 <KeywordSearch
                   label={searchLabel}
+                  searchOnAutcompleteSelection={landingPageCheck}
```
Comment (zurbergram) [inline src/applications/gi/containers/LandingPage.jsx:181]:
```suggestion
                  searchOnAutcompleteSelection
```

I think just something like this is valid

**Review by zurbergram — APPROVED** [review]

File: src/applications/gi/containers/LandingPage.jsx (hunk 3 above)
Comment (ksantiagomorgan6) [reply src/applications/gi/containers/LandingPage.jsx]: updated

**Review by sonntag-adam — APPROVED** [review]

**Pull request merged**